### PR TITLE
Remove compileElements and projectInternalView API

### PR DESCRIPTION
**Pull request by big-guy**

### CODE REVIEW

**Comment by big-guy** [comment]
@bot-gradle test this

File: subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ProjectInternalViewDependencyIntegrationTest.groovy [hunk 1]
```diff
@@ -17,10 +17,12 @@
 package org.gradle.api.plugins
 
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import spock.lang.Ignore
 
 /**
  * Tests that {@code JvmComponentDependencies#projectInternalView()} behaves as expected.
  */
+@Ignore("removed from 8.0")
```
Comment (jvandort) [inline subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ProjectInternalViewDependencyIntegrationTest.groovy:25]: Why not just delete this class?

**Review by jvandort — APPROVED** [review]

File: subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ProjectInternalViewDependencyIntegrationTest.groovy (hunk 1 above)
Comment (big-guy) [reply subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ProjectInternalViewDependencyIntegrationTest.groovy:25]: I added this when I thought I was going to be ignoring more tests vs changing/deleting

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Pull request merged**